ibis/expr/operations.py

```diff
@@ -1966,7 +1966,16 @@ def schema(self):
         names = []
 
         for projection in self.selections:
-            if isinstance(projection, ir.ValueExpr):
+            if (
+                isinstance(projection, ir.StructValue)
+                and not projection.has_name()
```

Comment:
why does the name matter?

Reply:
Also added `test_elementwise_udf_struct` to add a named struct column

Reply:
hmm, is this the api we really want, e.g. this implicit naming behavior? it seems sensible but no easy way to turn this on/off or error check it.